### ISSUE TITLE
cr_expect_arr_eq_cmp assertion issue

### ISSUE THREAD

**Issue by arthurteisseire** [Snaipe/Criterion#269]
Hi,

I'm using this test :
```
static int cmp(int **a, int **b)
{
        if (**a == **b)
                return (0);
        return (1);
}

Test(simple, test)
{
        int nb = 0;
        int *src[] = {&nb, NULL};
        int *dest[2];

        dest[0] = src[0];
        dest[1] = src[1];
        cr_assert_arr_eq_cmp(dest, src, sizeof(int *), cmp);
}
```
This test crash.
According to the documentation, the argument 'size' or this assertion function is the **number of bytes to check**.
I think that this argument is checking the number of adresses or 8 bytes for 64 bits systems instead of 1 byte.
To prove it I changed this line
`cr_assert_arr_eq_cmp(dest, src, sizeof(int *), cmp);`
into
`cr_assert_arr_eq_cmp(dest, src, 1, cmp);`
The assertion successed and it didn't crashed.
Then I changed the same line into
`cr_assert_arr_eq_cmp(dest, src, 2, cmp);`
It crashed. In fact I'm dereferencing a null pointer here.

Thanks for reading and for sharing this lib

**Comment by Snaipe** [comment]
Ah, yes, this is an error in the documentation. `Size` here should really mean `Length` instead.

**Comment by Snaipe** [comment]
This is also somewhat unfortunate, because the docs won't update unless I tag a release, and the new release will introduce the new assertion API, which deprecates those. Short of actually fixing that, I think
we'll have to keep this open as a heads up for people coming here with this exact problem.

**Comment by Snaipe** [comment]
This will be fixed in the upcoming v2.4.0 release docs

**Comment by interrupt-routine** [comment]
@Snaipe 
The problem is still there or reappeared in the old assertions reference:
https://criterion.readthedocs.io/en/master/assert_old.html#c.cr_assert_arr_eq_cmp

All of the `cr_assert_arr_*_cmp()` incorrectly have:
> Size – [in] Number of bytes to check
